lib/SILOptimizer/Transforms/OptRemarkGenerator.cpp

```diff
@@ -101,8 +101,11 @@ class OptRemarkGenerator : public SILFunctionTransform {
   bool isOptRemarksEnabled() {
     // TODO: Put this on LangOpts as a helper.
     auto &langOpts = getFunction()->getASTContext().LangOpts;
+
+    // If we have a remark streamer, emit everything.
     return bool(langOpts.OptimizationRemarkMissedPattern) ||
```

Comment:
@francisvm This whole predicate seems a little odd. Seems like we can clean up a bunch of checks in the emitter itself if it had a direct reference to the module's remark streamer instead of the module itself.

Reply:
The only issue I see is that the remark streamer is only used to serialize remarks, while the emitter is used to emit both diagnostics and serialize remarks.